lib/charms/mongodb/v1/helpers.py

```diff
@@ -91,14 +92,14 @@ def get_mongos_args(
     """
     # mongos running on the config server communicates through localhost
     # use constant for port
-    config_server_uri = f"{config.replset}/localhost:27017"
+    config_server_db = config_server_db or f"{config.replset}/localhost:27017"
 
     full_conf_dir = f"{MONGODB_SNAP_DATA_DIR}{CONF_DIR}" if snap_install else CONF_DIR
     cmd = [
         # mongos on config server side should run on 0.0.0.0 so it can be accessed by other units
         # in the sharded cluster
         "--bind_ip_all",
```

Comment:
Question

MongoDB has fancy interals so this also sets up a unix socket :
```
ubuntu@juju-8c71f6-1:~$ ss -l | grep -i mongo
u_str LISTEN 0      4096                                      /tmp/mongodb-27018.sock 623618975                       * 0
```

Should we restict it to **only** allow unix socket connections?

Reply:
for config server `--bind_ip_all` is better.
for subordinate charm - only socket